packages/firebase_in_app_messaging/example/lib/main.dart

```diff
@@ -0,0 +1,122 @@
+import 'dart:async';
+
+import 'package:firebase_analytics/firebase_analytics.dart';
+import 'package:firebase_analytics/observer.dart';
+import 'package:firebase_in_app_messaging/firebase_in_app_messaging.dart';
+import 'package:flutter/material.dart';
+
+void main() => runApp(MyApp());
+
+class MyApp extends StatelessWidget {
+  static FirebaseAnalytics analytics = FirebaseAnalytics();
+  static FirebaseAnalyticsObserver observer =
+      FirebaseAnalyticsObserver(analytics: analytics);
+  static FirebaseInAppMessaging fiam = FirebaseInAppMessaging();
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+        home: Scaffold(
+      appBar: AppBar(
+        title: const Text('In-App Messaging example'),
+      ),
+      body: Builder(builder: (BuildContext context) {
+        return Center(
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+            children: <Widget>[
+              AnalyticsEventExample(),
+              ProgrammaticTriggersExample(fiam),
+            ],
+          ),
+        );
+      }),
+    ));
+  }
+}
+
+class ProgrammaticTriggersExample extends StatelessWidget {
+  const ProgrammaticTriggersExample(this.fiam);
+
+  final FirebaseInAppMessaging fiam;
+
+  @override
+  Widget build(BuildContext context) {
+    return Card(
+      child: Padding(
+        padding: const EdgeInsets.all(24.0),
+        child: Column(
+          children: <Widget>[
+            Text(
+              "Programmatic Trigger",
+              style: TextStyle(
+                fontStyle: FontStyle.italic,
+                fontSize: 18,
+              ),
+            ),
+            const SizedBox(height: 8),
+            const Text("Manually trigger events programmatically "),
+            const SizedBox(height: 8),
+            RaisedButton(
+              onPressed: () async {
+                await fiam.triggerEvent('chicken_event');
```

Comment:
Is awaiting the event really necessary here? If calling `await` isn't something we want developers to feel obligated to do, then perhaps we should have `onPressed` be synchronous here and just fire and forget `triggerEvent`. There doesn't seem to be any error handling on the native side, so triggerEvent is unlikely to fail, and I think the Future<void> error handling is mostly just for use in tests.

Reply:
Good call - done!